napari/layers/_layer_actions.py

```diff
@@ -37,7 +37,12 @@ def _duplicate_layer(ll: LayerList):
 
     for lay in list(ll.selection):
         new = deepcopy(lay)
-        new.name += ' copy'
+        # new.events.disconnect()
+        new.events.source = new
+        for emitter in new.events.emitters.values():
+            # emitter.disconnect()
+            emitter.source = new
+        new.name += trans._(' copy')
```

Comment:
Nit: to handle all languages (especially with respect to right-to-left scripts) you probably want to do this instead.

```suggestion
        new.name = trans._('{name} copy', name=new.name)
```